practice/caesar/site.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+/**
+ * Returns string encrypted by moving each letter ahead in alphabet by key.
+ *
+ * Assumes characters are upper-case.
+ * All non-letters are preserved as-is (assumming they are punctuation)
+ *
+ * @type {String}
+ */
+function caesarEncrypt(plainStr, key) {
+  var encryptedText = '';
+  for (var i = 0; i < plainStr.length; i += 1) {
+    var charVal = plainStr[i].charCodeAt();
+    var aCharCode = 'A'.charCodeAt();
+    var zCharCode = 'Z'.charCodeAt();
+    if (charVal >= aCharCode && charVal <= zCharCode) {
+      var encCode = (charVal - aCharCode + key) % 26 + aCharCode;
+      encryptedText += String.fromCharCode(encCode);
+    } else {
+      encryptedText += plainStr[i];
+    }
+  }
```

Comment:
This problem is a pretty clean "mapping" where each letter is rotated in isolation. Practice using Lodash's map and string join to isolate that sub-problem.